source/js/components/search-form.js

```diff
@@ -0,0 +1,108 @@
+import fetchUtils from "@degjs/fetch-utils";
+
+const searchForm = () => {
+
+
+    const formEl = document.querySelector('.js-search-form');
+    const layoutEl = document.querySelector('.js-general-content');
+    const searchFieldSet = document.querySelector('.js-search-fieldset');
+    let searchInputEl;
+    const searchButtonEl = formEl.querySelector('.js-search-btn');
+
+    const url = formEl.dataset.api;
+
+    function init() {
+        searchInputEl = formEl.querySelector('.js-search');
+        formEl.addEventListener('submit', searchSubmit);
+    }
+
+    function searchSubmit(e) {
```

Comment:
[suggestion] This is nitpicky, but for event callback method names, consider adding some sort of action word to its name to help indicate what it is, such as `onSearchSubmit` or `handleSearchSubmit`.